mod-source-record-storage-server/src/main/java/org/folio/dao/PostgresClientFactory.java

```diff
@@ -1,13 +1,33 @@
 package org.folio.dao;
 
-import io.vertx.core.Vertx;
+import java.util.HashMap;
+import java.util.Map;
+
 import org.folio.rest.persist.PostgresClient;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.pgclient.PgConnectOptions;
+import io.vertx.pgclient.PgPool;
+import io.vertx.sqlclient.PoolOptions;
+
 @Component
 public class PostgresClientFactory {
 
+  private static final String HOST = "host";
+  private static final String PORT = "port";
+  private static final String DATABASE = "database";
+  private static final String PASSWORD = "password";
+  private static final String USERNAME = "username";
+
+  private static final String DEFAULT_SCHEMA_PROPERTY = "search_path";
+
+  private static final int POOL_SIZE = 5;
+
+  private static final Map<String, PgPool> pool = new HashMap<>();
```

Comment:
Can it be cunccurentHashMap and "private synchronized PgPool getPool(String tenantId)" make static without synchronized?

Reply:
or just method without synchronized

Reply:
If map is concurrent and method is not synchronized, it would be possible to have two `PgPool` being created at the same time and the one that completes second would replace the first put in the map. We currently cannot make the method static as still require the vertx injected bean. I think an issue is justified to create `PgPool` without `PostgresClient` to retrieve database connection config. Then we can simplify and control the map better.

Reply:
ok, my main concern that synchronized methods can slowdown vertx event loop. So we need to think about the async approach or figure out way to deal with it after RMB updates. Can you create a task to cover this issue?

Reply:
Yes, I will create the issue for asynchronous access to get client. Should there be another for getting connection configuration without PostgresClient?

Reply:
Done. [MODSOURCE-138](https://issues.folio.org/projects/MODSOURCE/issues/MODSOURCE-138)